src/renderer/components/inputs/SliderInput.tsx

```diff
@@ -117,15 +171,14 @@ const SliderInput = memo(
                         precision={precision}
                         size="xs"
                         step={controlsStep}
-                        value={formatValString(String(sliderValue ?? def))}
-                        onChange={(_, v) => {
-                            setInput(Math.min(Math.max(v, min), max));
-                        }}
+                        value={addUnit(inputString)}
+                        onBlur={onBlur}
+                        onChange={onNumberInputChange}
                     >
                         <NumberInputField
                             m={0}
                             p={1}
-                            w="3.1rem"
+                            w={`${dynamicNumInputWidth}rem`}
```

Comment:
Now sliders with different units will be misaligned. I think it would be easier to just set to "5em" or something.

Reply:
5 was the first thing I tried, and I didn't like how much it crowded out the slider with negative space. Your point is a good one though, so I've changed it so a constant amount is added if a unit is present. The precision is still a dynamic factor though, so sliders won't have all that negative space. We can always tweak the unit constant later. Right now it is enough to accommodate mm or three narrower characters.

Reply:
I tried that initially, but really didn't like how much it crowded the sliders with negative space and made them smaller, which impedes their controllability. Your point is a good one though. For now, I'm keeping precision dynamic (I don't *expect* to have one node with sliders of different precision, but we can always revisit this if the situation arises), but I've made it so a static amount is added if unit is present. My admittedly a little hacky solution to one slider having units and another not is to just added unit="" (which does not impede number deletion like characters currently do, until that's fixed) to sliders without units. It's not a beautiful solution, but I think it's preferable to having a lot of negative space in slider nodes from a fixed width.